### ISSUE TITLE
Cannot update Docker Desktop on Windows from 4.5.1 (74721) to 4.6.1 (76265)

### ISSUE THREAD

**Issue by Migrain** [docker/for-win#12633]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID:

### Actual behavior
Installation does not complete and in windows log there is this:

Application: update-1438851335.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.UnauthorizedAccessException
   at System.Security.AccessControl.Win32.SetSecurityInfo(System.Security.AccessControl.ResourceType, System.String, System.Runtime.InteropServices.SafeHandle, System.Security.AccessControl.SecurityInfos, System.Security.Principal.SecurityIdentifier, System.Security.Principal.SecurityIdentifier, System.Security.AccessControl.GenericAcl, System.Security.AccessControl.GenericAcl)
   at System.Security.AccessControl.NativeObjectSecurity.Persist(System.String, System.Runtime.InteropServices.SafeHandle, System.Security.AccessControl.AccessControlSections, System.Object)
   at System.Security.AccessControl.NativeObjectSecurity.Persist(System.String, System.Security.AccessControl.AccessControlSections, System.Object)
   at System.Security.AccessControl.FileSystemSecurity.Persist(System.String)
   at Docker.Core.PermissionUtils.SetAdministratorLocalSystemOnly(Docker.Core.ILogger, System.String)
   at Docker.Core.PermissionUtils.RemoveGlobalWrite(Docker.Core.ILogger, System.String)
   at CommunityInstaller.Program.Main(System.String[])

Starting Docker desktop the version remains at 4.5.1 (74721)

### Expected behavior
Docker installation updated to complete and updated to version 4.6.1 (76265)
### Information

  - Windows Version: Windows 10 Enterprise, 20H2, 19042.1586
  - Docker Desktop Version:  4.5.1
  - WSL2 or Hyper-V backend?  WSL2
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: no

### Steps to reproduce the behavior

  1. Press Download update from inside Docker Desktop
  2. restart to update
  3. Installation asking for permission to run 
  4. Pressing ok then nothing happens

**Comment by cbusquet** [comment]
Thank you ydelattre for your analysis, C:\ProgramData\DockerDesktop was the problematic folder for me.
I have been stuck on 4.5.1 hoping that a future update would solve my problem and managed to update to 4.7.1 today.

I did not remove my previous installation, I just changed the rights of the C:\ProgramData\DockerDesktop folder to full control as admin (I only had the read and write rights), then I relaunched the installer. After the installation everything worked fine.
